src/controllers/loginController.ts

```diff
@@ -0,0 +1,66 @@
+import { Request, Response } from 'express';
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+import passport from 'passport';
+import { Strategy as LocalStrategy } from 'passport-local';
+import Database from '../database';
+import { sendResponse } from '../utils';
+import config from '../config/config';
+
+/// ///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// eslint-disable-next-line require-jsdoc
+export async function checkUserCredentials(
+  email: string,
+  password: string,
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  done: Function
+) {
+  try {
+    // checking if user with this email exists in database
+
+    const user = await Database.User.findOne({
+      where: { email },
+    });
+
+    // comparing password entered with password stored in database
+
+    if (!user || !(await bcrypt.compare(password, user.password))) {
+      return done(null, false, { message: 'User not found' });
+    }
+    return done(null, user);
+  } catch (err) {
+    return done(err);
+  }
+}
+
+passport.use(
+  new LocalStrategy({ usernameField: 'email' }, checkUserCredentials)
+);
+
+const validatingUser = async (req: Request, res: Response) => {
+  passport.authenticate(
+    'local',
+    { session: false },
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    async (err: any, user: any) => {
+      // test if uer not found
+
+      if (!user) return sendResponse<null>(res, 400, null, 'User not found');
+
+      // generate token
+
+      const { secret } = config();
+      const token = jwt.sign({ id: user.id }, secret, { expiresIn: '2h' });
+
```

Comment:
I told you to separate this with current codes

Reply:
The only way I did this is to remove that strategy and put it in app.ts.Could you explain further?